cairo-contracts/packages/contracts/src/tests/client.cairo

```diff
@@ -207,3 +208,281 @@ fn test_client_recover_active_client() {
+    let updating_time = cfg.latest_timestamp.clone() + TIMESTAMP(1);
+
+    // Create a `MsgUpdateClient` message.
+    let msg = cfg
+        .dummy_msg_update_client(
+            subject_client.client_id.clone(),
+            subject_client.height.clone(),
+            third_updating_height,
+            updating_time.clone(),
+        );
+
+    // Submit a `MsgUpdateClient` to the IBC core contract.
+    core.update_client(msg.clone());
+
+    // -----------------------------------------------------------
+    // Wait for 10 seconds
+    // -----------------------------------------------------------
+
+    let new_timestamp = cfg.latest_timestamp.clone().as_secs() + 10;
+    start_cheat_block_timestamp_global(new_timestamp);
+    start_cheat_block_number_global(5);
+
+    cfg.latest_timestamp = (new_timestamp * 1_000_000_000).into();
+    cfg.latest_height.revision_height = cfg.latest_height.revision_height + 5;
+
+    // -----------------------------------------------------------
+    // Check consensus states have not yet been pruned
+    // -----------------------------------------------------------
+
+    let first_consensus_state = comet.consensus_state_root(0, second_updating_height.clone());
+    assert!(first_consensus_state.is_non_zero());
+    let second_consensus_state = comet.consensus_state_root(0, second_updating_height.clone());
+    assert!(second_consensus_state.is_non_zero());
+    let third_consensus_state = comet.consensus_state_root(0, third_updating_height.clone());
+    assert!(third_consensus_state.is_non_zero());
+
+    // -----------------------------------------------------------
+    // Wait timeout and retrieve status
+    // -----------------------------------------------------------
+
+    let new_timestamp = cfg.latest_timestamp.clone().as_secs() + 101;
+    start_cheat_block_timestamp_global(new_timestamp);
+    start_cheat_block_number_global(5);
+    assert!(comet.status(0).is_expired());
+
+    // -----------------------------------------------------------
+    // Recover client
+    // -----------------------------------------------------------
+
+    cfg.latest_timestamp = (new_timestamp * 1_000_000_000).into();
+    cfg.latest_height.revision_height = cfg.latest_height.revision_height + 5;
+    let substitute_client = cfg.create_client(@core);
+
+    // Create a `MsgRecoverClient` message.
+    let msg = cfg.dummy_msg_recover_client(subject_client.client_id, substitute_client.client_id);
+
+    // Submit a `MsgRecoverClient` to the IBC core contract.
+    let _recover_resp = core.recover_client(msg.clone());
+
+    // -----------------------------------------------------------
+    // Check Results
+    // -----------------------------------------------------------
+
+    assert!(comet.status(0).is_active());
+
+    let third_consensus_state = comet.consensus_state_root(0, third_updating_height.clone());
+    assert!(third_consensus_state.is_non_zero());
```

Comment:
why this should be non_zero? this should be pruned too.

Reply:
Oh this was from before, when we were keeping latest height, will correct